purchases/src/customEntitlementComputation/kotlin/com/revenuecat/purchases/PurchasesConfiguration.kt

```diff
@@ -0,0 +1,105 @@
+package com.revenuecat.purchases
+
+import android.content.Context
+import java.util.concurrent.ExecutorService
+
+open class PurchasesConfiguration(builder: Builder) {
+
+    val context: Context
+    val apiKey: String
+    val appUserID: String
+    val observerMode: Boolean
+    val service: ExecutorService?
+    val store: Store
+    val diagnosticsEnabled: Boolean
+    val dangerousSettings: DangerousSettings
+    val verificationMode: EntitlementVerificationMode
+
+    init {
+        this.context = builder.context
+        this.apiKey = builder.apiKey
+        this.appUserID = builder.appUserID
+        this.observerMode = builder.observerMode
+        this.service = builder.service
+        this.store = builder.store
+        this.diagnosticsEnabled = builder.diagnosticsEnabled
+        this.verificationMode = builder.verificationMode
+        this.dangerousSettings = builder.dangerousSettings
+    }
+
+    open class Builder(
+        @get:JvmSynthetic internal val context: Context,
+        @get:JvmSynthetic internal val apiKey: String,
+        @get:JvmSynthetic internal val appUserID: String,
+    ) {
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var observerMode: Boolean = false
+
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var service: ExecutorService? = null
+
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var store: Store = Store.PLAY_STORE
+
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var diagnosticsEnabled: Boolean = false
+
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var verificationMode: EntitlementVerificationMode = EntitlementVerificationMode.default
+
+        // TODO Default to custom entitlement computation mode
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var dangerousSettings: DangerousSettings = DangerousSettings()
+
+        fun observerMode(observerMode: Boolean) = apply {
+            this.observerMode = observerMode
+        }
+
+        fun service(service: ExecutorService) = apply {
+            this.service = service
+        }
+
+        fun store(store: Store) = apply {
+            this.store = store
+        }
+
+        /**
+         * Enabling diagnostics will send some performance and debugging information from the SDK to our servers.
+         * Examples of this information include response times, cache hits or error codes.
+         * This information will be anonymized so it can't be traced back to the end-user.
+         * The default value is false.
+         */
+        fun diagnosticsEnabled(diagnosticsEnabled: Boolean) = apply {
```

Comment:
I'm wondering, should we disable diagnostics in `customEntitlementComputation` mode, @aboedo?

Reply:
let's keep it disabled on v1 and we can take it from there